### PR TITLE
Fix bundle #4

### LINKED ISSUE #7103: Light Floors are not affected by shadowling abilities
Shadowlings can't disable light floors with their abilities and get screwed over by light floors covered by glass.

### LINKED ISSUE #14270: Machine people can drink from condiment bottles
**Issue Description**:
Machine people can drink from condiment bottles

**What did you expect to happen**: 
As when eating food, a message that you have no mouth.

**What happened instead**:
You drink the condiment.

**Why is this bad/What are the consequences**:
Machine people aren't supposed to be able to eat.

**Steps to reproduce the problem**:
Create or find a condiment bottle.  Use it on yourself.

**When did the problem start happening**:
Unknown.

**Extra information**:

### LINKED ISSUE #14276: Security jacket is invisible for greys.
**Issue Description**:
Grey's security jacket sprite is now invisible.

**What did you expect to happen**: 
To see the security jacket on my character when I put it on.

**What happened instead**:
When I put on the security jacket, the sprite doesnt appear on my character

**Why is this bad/What are the consequences**:
Makes greys unable to use the alternative to the armor, the security jacket.

**Steps to reproduce the problem**:
Be a grey. and then put on a security jacket

**When did the problem start happening**:
After the grey sprite update, where the armor and hailer were fixed, probably something there broke it.

**Extra information**:
Screenshots of the jacket not showing
image
image

### LINKED ISSUE #14312: Possessed Blade pAI Alert Mismatch
**Problem Description**:
When the Chaplain calls for someone to possess their possessed blade, it throws up a pAI available alert in the chat. The icon in the screen itself shows the blade, however.

**What did you expect to happen**: 
When the Chaplain does that, it should say they are looking for a spirit for their blade in the chat.

**What happened instead**:
The prompt states pAI, but when you hit teleport it'll bring you to the chaplain and his blade.

**Why is this bad/What are the consequences:**
Someone unthinkingly could take position in the blade, which is not what they clicked on. 

**Steps to reproduce the problem**:
1. Have the Chaplain take the possessed blade.
2. Call for ghosts.
3. Chat shows a wrong prompt.

**When did the problem start happening**:
Just today, though I'm unsure how long it has been happening.

**Possibly related stuff (which gamemode was it? What were you doing at the time? Was anything else out of the ordinary happening?)**:
This was during a testmerge, so I can't say for sure if this happens every round. However I doubt the TM affected this specific scenario.

### LINKED ISSUE #14313: The item 'Spooky Lantern' does not allow the user to have thermal vision when resummoning the wisp.
**Problem Description**:
The item 'Spooky Lantern' does not allow the user to have thermal vision again when re-summoning the wisp.

**What did you expect to happen**: 
The item 'Spooky Lantern' should enable the user to have thermal vision as long as the wisp is released from the lantern, regardless of the amount of time it gets summoned.

**What happened instead**:
The user is able to gain thermal vision the first time they released the wisp. The problem begins when the user returns the wisp back to the lantern, and then releases the wisp again, where thermal vision is not gained back for the user. The 'Spooky Lantern' also does not provide any additional increase in light.

**Why is this bad/What are the consequences:**
The 'Spooky Lantern' item is not consistent with its mechanic or at least does not indicate to the user that the lantern is only a one-time use. This will disrupt the user experience or immersion during gameplay.  

**Steps to reproduce the problem**:
1. Use Byond client version 1528.
2. Proceed to find 'Spooky Lantern' in Lavaland.
3. Go to a populated area and hide behind a wall from the population.
4. Release the wisp and observe that the mechanic is working as intended.
5. Return the wisp, and re-release the wisp.
6. Observe that the 'Spooky Lantern' mechanic does not work as intended, and cannot be resolved.

Addendum:
1. This issue can be resolve if the user reconnects the game whilst the 'Spooky Lantern' is in use. However, the problem still persist if the user performs Step 5 again.
2. This issue is not resolved if the user reconnects the game whilst the 'Spooky Lantern' is not in use. Releasing the wisp afterwards does not work.

**When did the problem start happening**:
This issue was discovered during a round where "Medical TGUI is being testmerged." I have not played the game in a while so I do not know when this problem exactly started.

**Possibly related stuff (which gamemode was it? What were you doing at the time? Was anything else out of the ordinary happening?)**:
This issue occured in a round where "Medical TGUI is being testmerged.".
scrnshot1
An image showing a populated area in the bar whilst the 'Spooky Lantern' is not turned on.

scrnshot12
An image showing the 'Spooky Lantern' does not provide thermal vision for the user to observe the fight inside the bar.

### LINKED ISSUE #14506: Free Golems drop items on exiting mech
**Issue Description**:
When a free golem exits a ripley, they drop all their items 

**What did you expect to happen**: 
to not lose the items

**What happened instead**:
lost the items

**Why is this bad/What are the consequences**:
have to pick everything back up again

**Steps to reproduce the problem**:
be a free golem and wear everything on your slots where the hands are

**When did the problem start happening**:
dont know, but recently

**Extra information**:

### LINKED ISSUE #14753: Being a cling disables Alt-clicking to examine a tile or to open a container
**Issue Description**:

Being a cling disabled my ability to alt-click a tile to examine it, or alt-click a container to open it. Upon removal of cling status, I was able to alt-click tiles/containters again.

**What did you expect to happen**: 

Alt-clicking a tile to examine it, or a container to open it.

**What happened instead**:

Nothing, alt-clicking did not open a container nor examine a tile.

**Why is this bad/What are the consequences**:

Being unable to use the hotkeys is pretty annoying.

**Steps to reproduce the problem**:

Started the round with cling enabled, became a cling, was unable to alt-click until I requested an admin remove my status as cling.

**When did the problem start happening**:

Had this problem playing a round on 27/10/20 at GMT 16:20

**Extra information**:

### LINKED ISSUE #14830: AI detector bug
**Issue Description**:
So, AI detector detects cryo'd AI, there was new AI, that was 99% not in the area (xeno lab), but the storaged AI was there before cryoing and my AI detector was on red 5 minutes while i waited. I even said provocative false message to make current AI look away, and judjung by chat it did.

**What did you expect to happen**: 
AI detector would'nt detect last position of cryo'd AI

**What happened instead**:
AI detector did detect last position of cryo'd AI

**Why is this bad/What are the consequences**:
I was confused, scared and wasted 5 minutes waiting in a locker for ai to go away

**Steps to reproduce the problem**:
Just walked around with uplink multitool i guess.

**When did the problem start happening**:
Everytime i was near xeno lab

**Extra information**:
round ended at 23:45 GMT+3 time (for logz)

### LINKED ISSUE #15218: Headcrabs can dupe zombies
**Issue Description**:
Headcrabs sometimes create 2 zombies from the same corpse.

**What did you expect to happen**: 
For every corpse to only spawn 1 zombie.

**What happened instead**:
2 identical looking zombies spawned from the same body. Only one of them drops the original corpse.

**Why is this bad/What are the consequences**:
This behaviour is really illogical and makes headcrabs more powerful than they should be.

**Steps to reproduce the problem**:
Spawn a ton of headcrabs, spawn in a corpse, should probably cause the issue as described above.

**When did the problem start happening**:
Probably for as long as headcrabs have been in the game.

**Extra information**:
According to @DoctorDrugs , this also happens with blob spores, but they couldn't reproduce it.

### LINKED ISSUE #16041: Colorblindness bug
**Issue Description**: Colorblindness, at least in vulps is messed up in a few situations i've seen. Particularly when the colossus uses the judgement verb and when the legion is released.

**What did you expect to happen**: For colorblindness not to be negated during these events.

**What happened instead**: while genetically it does not remove the colorblindness, the flash of red resets the screen and makes everything in full color again.

**Why is this bad/What are the consequences**: it breaks immersion, particularly when that is a major part of the character, unless it is being done through genetics.

**Steps to reproduce the problem**: Have colossus attack and use the "Judgement" attack, release legion, vulp with colorblindness. Not tested on other species with colorblindness, only noticed it as I mostly play vulp.

**When did the problem start happening**: A couple days ago when we had the BYO colony on lavaland, someone released legion, both myself and another vulp player lost colorblindness. Again today, colossus was released on the station and i lost colorblindness when it used "judgement"

**Extra information**: Not much else other than it seems to be whenever that red filter appears on the screen, colorblindness seems to not understand it and simply resets the screen, but doesnt mess with the genetics.

### LINKED ISSUE #16279: Vulp mimes make noise when growling
**Issue Description**:

Vulp mimes growl and make the noise instead of growling silently

**What did you expect to happen**: 

x acts out a growl

**What happened instead**:

x growls (and noise plays)

**Why is this bad/What are the consequences**:

mime make noise

**Steps to reproduce the problem**:

be vulp
take vow of silence
growl

**When did the problem start happening**:

probably been there

**Extra information**:

you act out a howl why not act out a growl

### LINKED ISSUE #16412: Revenants aren't listed under antags in the orbit menu with AHUD on
**Issue Description**:
Revs aren't in the list

**What did you expect to happen**: 
Revs to be in the list

**What happened instead**:
Rev wasn't in the list

**Why is this bad/What are the consequences**:
Bugs = bad

**Steps to reproduce the problem**:
1. Have a revenant
2. Enable AHUD
3. Open orbit menu and see that revenant isn't on the list

**When did the problem start happening**:

**Extra information**:
Screenshot (456)

### LINKED ISSUE #16438: You can teleport signs to you when screwing them in place
**Issue Description**:
When you try to screw a sign in place (such as warning signs or the department direction ones), you can walk away with the "select direction" prompt open and the sign'll teleport to you when you finally pick a direction, even if it's far away or in someone's hands.

**What did you expect to happen**: 
Nothing, or an error message similar to "You're too far away from the sign!"

**What happened instead**:
Teleporting signs.

**Why is this bad/What are the consequences**:
Looks jank, is a bug. I don't know if there's a way to abuse this for powergaming but it is still bad.

**Steps to reproduce the problem**:
First get a sign or detack one from a wall/window, then:

1. Click the sign with a screwdriver in hand to open the Select direction window.
2. Click back into the game window to regain focus and walk away with the direction menu still open
3. Select a direction while being away from the sign.
image

**When did the problem start happening**:
If I had to guess it might be an old bug, but no idea

**Extra information**:
I can't help but wonder what happen if two people do this simultaneously, but I can't really test it myself.

### LINKED ISSUE #16466: Pink hawaiian shirt has no drask sprite
**Issue Description**:

When wearing a pink hawaiian shirt as a drask, the sprite is missing. This is not a problem for the other shirts.

**What did you expect to happen**: 

To have a drippin pink shirt on.

**What happened instead**:

Dripless. There is no sprite. Just some red text above my head.

**Why is this bad/What are the consequences**:

I cannot have pink hawaiian shirt drip. And missing sprites are just bad.

**Steps to reproduce the problem**:

1. Be drask
2. Put on a pink hawaiian shirt
3. Cry

**When did the problem start happening**:

No idea, might have never worked.

**Extra information**:

simply gamebreaking bug

### LINKED ISSUE #16471: Item Teleportation
**Issue Description**:
Items can be teleported using wrapping paper if you wrap the same item twice really quickly.

**Why is this bad/What are the consequences**:
You can potentially do some crazy stuff with this as an antag...

**Steps to reproduce the problem**:
Click twice very quickly on an item with wrapping paper, two packages are created. Unwrap one package, drop the item, move away, unwrap the other package. This teleports the item to your feet.

**When did the problem start happening**:
Unknown.

### LINKED ISSUE #16508: Using changeling epinephrine overdose to wake up from cryo makes you permanently blind
**Issue Description**:

Using the epinephrine overdose power to wake up early from cryo healing makes you permanently blind.

**What did you expect to happen**: 

The blindness to go away when I woke up.

**What happened instead**:

An admin had to remove my blindness

**Why is this bad/What are the consequences**:

Makes you perma blind for trying to wake up faster from cryo.

**Steps to reproduce the problem**:

1. Get hurt.
2. Enter cryo.
3. When auto-ejected, use epinephrine overdose to wake up faster.
**When did the problem start happening**:
I noticed it a day or two ago.

### PR DESCRIPTION
**Pull request by SabreML**
## What Does This PR Do

Please see the changelog for a list, because there's no need to type all of that again.
Each fix is in a separate commit for ease of review.

Fixes #16508.
Fixes #16471.
Fixes #16466.
Fixes #16438.
Fixes #16412.
Fixes #16279.
Fixes #16041.
Fixes #14312.
Closes #15218. (Headcrabs were removed.)
Closes #14830. (Fixed in #15535.)
Closes #14753. (Seems to be a user error.)
Closes #14506. (Fixed in #16338.)
Closes #14313. (Fixed in #16411.)
Closes #14276. (Seems to have been fixed.)
Closes #14270. (IPCs can drink reagents.)
Closes #7103. (Fixed in #14316.)
(I *think* that's all of them.)

## Changelog
:cl:
add: Added Blob-infected players and mice to the 'Antagonist' orbit menu if AntagHUD is enabled.
add: Added Revenants to the 'Antagonist' orbit menu even if AntagHUD is disabled.
tweak: Made progress bars show above regular HUD elements.
tweak: Added ant-repellant to the wrappers of vending machine food.
tweak: Made Vulpkanin mimes growl silently.
tweak: Made the Possessed Blade ghost poll actually say 'Possessed Blade', rather than 'pAI'.
fix: Fixed 'APC Access ID cards' not having a sprite set.
fix: Fixed being able to pat out a fire on yourself.
fix: Fixed gibbers vanishing after being deconstructed.
fix: Fixed using 'Epinephrine Overdose' as a changeling to wake up causing permanent blindness in some circumstances.
fix: Fixed package wrappers being able to wrap the same package multiple times.
fix: Fixed being able to teleport items around the station using the above package wrappers.
fix: Fixed Drask being unable to wear pink Hawiian shirts.
fix: Fixed being able to teleport station signs by using a screwdriver on them.
fix: Fixed various vending machines causing runtimes when reconstructed.
fix: Fixed hybrid turret inhands being flipped.
fix: Fixed some screen colour flash effects during megafauna fights acting as a cure for colourblindness.
fix: Fixed disabling night shift mode in the config not working.
spellcheck: Replaced "Targetted" with "Targeted" in the rogue drone event announcement.
/:cl:

### CODE REVIEW

File: code/modules/awaymissions/mission_code/ruins/oldstation.dm [hunk 1]
```diff
@@ -43,7 +43,7 @@
 /obj/item/card/id/away/old/apc
 	name = "APC Access ID"
 	desc = "A special ID card that allows access to APC terminals."
-	icon_state = "centcom_old"
+	icon_state = "guest_invalid"
```
Comment (SabreML) [inline code/modules/awaymissions/mission_code/ruins/oldstation.dm:46]: To clarify for this one, `centcom_old` isn't an icon state anymore so it just showed the entire file when you examined it.

**Comment by hal9000PR** [comment]
#7103 isn't an issue anymore as shadowlings can disable light tiles.

**Review by variableundefined — APPROVED** [review]

**Comment by variableundefined** [comment]
@SabreML in the future, please break down fixes into much more atomic PR, like 2 - 3 at once, since massive fix PR are hard to test and get delayed easily.

**Comment by SabreML** [comment]
> @SabreML in the future, please break down fixes into much more atomic PR, like 2 - 3 at once, since massive fix PR are hard to test and get delayed easily.

@variableundefined Ah alright, sorry about that. I was trying to reduce the number of PRs that would be needed, since in this case it'd take quite a lot of them if it was kept properly atomic.

**Pull request merged**